src/com/google/cose/CoseKey.java

```diff
@@ -128,6 +128,19 @@ void verifyAlgorithmMatchesKey(Algorithm algorithm) throws CborException, CoseEx
     }
   }
 
+  public static CoseKey generateKey(Algorithm algorithm) throws CborException, CoseException {
```

Comment:
I don't think `Algorithm` is actually the right input here.

Instead, it makes more sense for the input to be something like "P-256" and we internally just match the hash to the signature size.

Additionally, this always assumes Ed25519, but we might want to support Ed448 some day.

Another option is expanding the `Algorithm` enum so that it includes key size as part of the definition. It's probably fine to assume "ECDSA with SHA-256" is P-256. However, assuming Ed25519 is the only edwards curve we ever support seems wrong to me.

Reply:
> Another option is expanding the Algorithm enum so that it includes key size as part of the definition.
This is what I imagined we'd do when we need to support different key sizes.

Reply:
I am not a cryptography expert, but I've added enums in Algorithm for ED25519 and ED448. Not sure what exactly to fill in for `coseAlgorithmId` and `javaAlgorithmId`, but I've left some filler.  The support for Ed448 key generation is probably a feature for the future.

Reply:
Oops, I just saw this comment here... I'll go resolve that comment so we can continue discussion here.

`-9` and `-10` are not valid algorithms, AFAICT.

These should match up with the IANA values: https://www.iana.org/assignments/cose/cose.xhtml#algorithms

Key size is not part of the algorithm. If we want to include key size, then we'll have to add a new field (IMO, this is the best way forward).

If you do add key size, then I think we'll want to remove `SIGNING_ALGORITHM_EDDSA`.

Reply:
+1, feel free to remove SIGNING_ALGORITHM_EDDSA in this change itself if you have already added enums for ED25519 and ED448 and replaced the usages.

Reply:
Looking at the IANA values, I can't find any values that match Ed25519 or Ed448 under COSE algorithms? It seems to be collectively grouped under EdDSA, value -8.

Reply:
That's because the algorithm is "EdDSA". The key size isn't actually part of the it --  at least it's not in the way that COSE defines "algorithm".

We have different algorithm ids for the various NIST curves because the algorithm is "curve with hash". Note that those algorithms don't define key size, either. See https://crypto.stackexchange.com/questions/19793/does-the-size-of-a-ecdsa-key-determine-the-hash-algorithm